chore(PPDSC-2412): audio player styling changes

mstuartf opened the pull request:
[PPDSC-2412](https://nidigitalsolutions.jira.com/browse/PPDSC-2412)

**What**

1. Background - why this is needed
- Audio player styling changes
2. What did you do
- Made styling changes
- Added codemod utility functions for updating nested props
- Added codemod to prevent visual changes from new version
- Added `...Old` style presets in order to override single style attributes. 
3. What does the reviewers should expect
- Audio player should match _new_ styling by default
- Running codemods should apply _old_ styling. 

<!---
This section will be used to indicate if we should move to a major version in the next release, remove if not revelant.
DO CONSIDER any of the following are breaking changes to a consumer, this is by no mean an exhaustive list.
-removing or renaming props
-removing or renaming tokens
-removing or renaming components
-removing or renaming exported functions
-in some cases, major bumps to peer dependencies
--->

<!---
Add any breaking change if present.
E.g:
BREAKING CHANGE: renames the foobar component's prop foo to bar
--->

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected


<!---
Below sections are optional
--->

**Before:**
<!--- Drag and Drop your screenshot's here --->

**After:**
<!--- Drag and Drop your screenshot's here --->

**Who should review this PR:**
<!---
If you know someone is a domain expert for your PR,
someone who is deeply involved in the story,
ask them explicitly to review the PR.
--->

**How to test:**
<!--
If it's not immediately obvious how to test this PR, give instructions.
It's mandatory to update README.MD or development documentation if existing test strategy had changed.
-->

<!--
More info about raising an good PR: https://nidigitalsolutions.jira.com/wiki/spaces/NPP/pages/1319370846/Pull+Request
-->


[PPDSC-2412]: https://nidigitalsolutions.jira.com/browse/PPDSC-2412?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

pp-serviceaccount commented:
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2412-styles
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2412-styles/storybook

JohnTParsons commented:
Should this be DO NOT MERGE until Dow Jones issue has been resolved? Would it be worth bringing back develop branch for breaking change releases (as an intermediate place to merge breaking change PRs to)?

Xin00163 reviewed (commented):
LGTM. It might be worth running codemod locally on DJ's repo which is using this component to see if there are any problems. I am trying to find the repo but looks like it's not used anywhere. @Vanals do you know which repo is using AudioPlayerComposable?

Vanals commented:
> LGTM. It might be worth running codemod locally on DJ's repo which is using this component to see if there are any problems. I am trying to find the repo but looks like it's not used anywhere. @Vanals do you know which repo is using AudioPlayerComposable?

Sounds good but no clue who is using it, designers **may** know.

JohnTParsons reviewed (approved):
Looks like a useful set of coded utility functions you have added

mstuartf commented:
> LGTM. It might be worth running codemod locally on DJ's repo which is using this component to see if there are any problems. I am trying to find the repo but looks like it's not used anywhere. @Vanals do you know which repo is using AudioPlayerComposable?

@Xin00163 apparently it is the WSJ repo that uses this component. Could we create an unstable release for the epic branch when all the tickets have been merged in and test on there, rather than doing each ticket individually?

Xin00163 commented:
> > LGTM. It might be worth running codemod locally on DJ's repo which is using this component to see if there are any problems. I am trying to find the repo but looks like it's not used anywhere. @Vanals do you know which repo is using AudioPlayerComposable?
> 
> @Xin00163 apparently it is the WSJ repo that uses this component. Could we create an unstable release for the epic branch when all the tickets have been merged in and test on there, rather than doing each ticket individually?

Yeah good idea

evgenitsn commented:
This is looking great. I have one question, what was the final agreement about how to implement these v7 component default updates in terms of codemods. Should it be one single codemod file where we will add all other components changes **or** we will have separate files for every component?

If it is the first one then might be a good idea to rename the `audio-player-default-styles` and `tests` to something more generic and to use these files for the future tickets. 

I am asking this because I am currently working on the Slider updates and I am not sure what will be the file structure. 

CC: @Xin00163 @Vanals @JohnTParsons

Xin00163 commented on src/audio-player-composable/components/volume-control/defaults.ts:
```diff
@@ -1,7 +1,7 @@
   audioPlayerVolumeControl: {
     transitionPreset: 'widthChange',
-    spaceBetween: 'space030',
+    spaceBetween: 'space010',
```
do we need to add this to codemod?

Xin00163 commented:
> This is looking great. I have one question, what was the final agreement about how to implement these v7 component default updates in terms of codemods. Should it be one single codemod file where we will add all other components changes **or** we will have separate files for every component?
> 
> If it is the first one then might be a good idea to rename the `audio-player-default-styles` and `tests` to something more generic and to use these files for the future tickets.
> 
> I am asking this because I am currently working on the Slider updates and I am not sure what will be the file structure.
> 
> CC: @Xin00163 @Vanals @JohnTParsons

I am conflicted. On one hand, it would be nice to run one codemod rather than >10; On the other hand, what if the user only want to revert one component's default values? Unless the codemod can take the component as an argument which I am not sure. As for name, I would though `revert-to-previous-component-defaults` something like that. We can have a discussion on Monday

evgenitsn commented:
> 



> I am conflicted. On one hand, it would be nice to run one codemod rather than >10; On the other hand, what if the user only want to revert one component's default values? Unless the codemod can take the component as an argument which I am not sure. As for name, I would though `revert-to-previous-component-defaults` something like that. We can have a discussion on Monday

Yeah, I agree or maybe another option would be to have them separate so the users can apply them one by one but to somehow join all of them in another command/file/something if they want to apply them all of once. 

- codemod-audio-player
- codemod-slider
- codemod-xxx

and a `"codemod-all-components"` which combines them if possible.

Agree, let's discuss on Monday.

mstuartf commented:
@Xin00163 @evgenitsn 

I've added a separate `all-default-styles` codemod that can import and apply multiple other codemods. When you add more, you should just be able to add them to the list that is iterated over. 

I've also updated the README to explain the structure. 

Does this fit what you had in mind?
